wreck: set CUDA_VISIBLE_DEVICES when gpus are in R_lite

grondo opened the pull request:
I thought I'd throw the current version of the CUDA_VISIBLE_DEVICES plugin into a PR.
This version also sets CUDA_DEVICE_ORDER as @dongahn instructs in #1598.

The per-task behavior can be requested with `-o gpubind=per-task` and the plugin can be disabled with `-o gpubind=off`.

Probably should figure out some way to test this in CI, but we first need a way to simulate gpu resources I guess...

SteVwonder commented:
Does this need modifications to `wreckrun` to handle the `gpubind` flag properly?
```
flux wreckrun -l -N1 -c10 -n4 -g1 -o gpubind=per-task printenv CUDA_VISIBLE_DEVICES
wreckrun: Failed to process cmdline args
```

grondo commented:
Oops, sorry I didn't push a fully functional PR (and I left debug code in the script)

Fixes coming shortly.

grondo commented:
Ok, this version should be better, though there is no CI testing yet.

grondo commented:
I've added support in `wreckrun` to fake gpus in `R_lite` if a scheduler is not loaded and the `-g, --gpus-per-task` option is used. 

This feature is then used to sanity check the `CUDA_VISIBLE_DEVICES` support added by the Lua plugin.

I also added the stanza to allow the `-o gpubind` to wreckrun/submit.

If travis and peer review passes, this should be ready to go in.

grondo commented:
Something in the new test isn't working in Travis. I'll have to run that down tonight.

grondo commented:
Hm, I guess my script-fu has failed though. Still failing in Travis but not on my other systems. :frowning_face:

grondo commented:
Oh, duh. Forgot to add `cuda_devices.lua` to Makefile.am.

garlick commented:
Great!  Merging.
